src/Symfony/Component/Asset/VersionStrategy/RemoteJsonManifestVersionStrategy.php

```diff
@@ -24,39 +24,23 @@
  *
  * You could then ask for the version of "main.js" or "css/styles.css".
  */
-class RemoteJsonManifestVersionStrategy implements VersionStrategyInterface
+class RemoteJsonManifestVersionStrategy extends AbstractJsonManifestVersionStrategy
 {
-    private $manifestData;
-    private $manifestUrl;
     private $httpClient;
 
     /**
      * @param string $manifestUrl Absolute URL to the manifest file
      */
-    public function __construct(string $manifestUrl, HttpClientInterface $httpClient)
+    public function __construct(string $manifestUrl, HttpClientInterface $httpClient, bool $strict = false)
     {
-        $this->manifestUrl = $manifestUrl;
+        parent::__construct($manifestUrl, $strict);
```

Comment:
this looks wrong to me. The parent class says it expects an absolute path, but you pass a URL here. That's not the same. This makes me thing that your extracted abstraction is the wrong one.

Reply:
I agree that this 1st argument has mixed contents and could accept even other things (like an array of manifest data). A `Manifest`. I think the `VersionStrategy` was not the best place for the manifests.

I'll give a try with a `ManifestPackage` class that accepts `Manifest` instances (implemented by `JsonManifest` and `RemoveJsonManifest`).

Reply:
I've updated the PR to stay in the scope of the feature. No more refactoring.